src/resolvers/git.cr

```diff
@@ -103,9 +103,19 @@ module Shards
     def self.normalize_key_source(key : String, source : String) : {String, String}
       case key
       when "git"
-        {"git", source}
+        uri = URI.parse(source)
+        downcased_host = uri.host.try &.downcase
+        if downcased_host && downcased_host.in?("github.com", "bitbucket.com", "gitlab.com")
```

Comment:
`www.` subdomains should also be included here.

Reply:
done